### PR TITLE
⚠️capk generates kubadm config files under /run/kubeadm/ instead of /tmp

### LINKED ISSUE #3769: CAPBK should not write to /tmp
https://cloudinit.readthedocs.io/en/latest/topics/modules.html#runcmd
the docs here say that cloud-init runcmd steps should not write to /tmp because of a race w/ systemd-tmpfiles-clean, but currently, CABPK writes to `/tmp/kubeadm.yaml` and `/tmp/kubeadm-join-config.yaml`. 

Instead, we should write to someplace like `/run/kubeadm`

/kind bug

### PR DESCRIPTION
**Pull request by arghya88**
**What this PR does / why we need it**:

cloud-init runcmd steps should not write to /tmp because of a race w/ systemd-tmpfiles-clean

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3769

### CODE REVIEW

File: test/infrastructure/docker/cloudinit/kindadapter_test.go [hunk 1]
```diff
@@ -133,10 +133,10 @@ write_files:
 		{Cmd: "mkdir", Args: []string{"-p", "/etc/kubernetes/pki"}},
 		{Cmd: "/bin/sh", Args: []string{"-c", "cat > /etc/kubernetes/pki/sa.key /dev/stdin"}},
 		{Cmd: "chmod", Args: []string{"0600", "/etc/kubernetes/pki/sa.key"}},
-		// /tmp/kubeadm.yaml
+		// /run/kubeadm/kubeadm.yaml
 		{Cmd: "mkdir", Args: []string{"-p", "/tmp"}},
```
Comment (benmoss) [inline test/infrastructure/docker/cloudinit/kindadapter_test.go:137]:
```suggestion
		{Cmd: "mkdir", Args: []string{"-p", "/run/kubeadm"}},
```

**Review by benmoss — CHANGES_REQUESTED** [review]
I don't think `/run/kubeadm` will exist by default and so might need to be created, unlike `/tmp`. Have you tested these changes?

File: test/infrastructure/docker/cloudinit/kindadapter_test.go (hunk 1 above)
Comment (arghya88) [reply test/infrastructure/docker/cloudinit/kindadapter_test.go]: Updated..Thanks

**Review by vincepri — APPROVED** [review]
/approve
/assign @fabriziopandini

**Comment by CecileRobertMichon** [comment]
@arghya88 please add a PR description describing why this change is desirable

**Comment by arghya88** [comment]
@CecileRobertMichon Thanks for feedback..I have added a description

**Comment by fabriziopandini** [comment]
Thanks @arghya88 !
PR looks good to me! My only concern is how do we start to document v1alpha4 breaking changes like this. Should we change the PR title as :warning ?
@vincepri ^^

**Comment by arghya88** [comment]
@fabriziopandini I am fine doing that if that's good enough. Have we done something else before to handle breaking change scenario?

**Comment by fabriziopandini** [comment]
Renaming the PR should be enough IMO, but let's wait for confirmation

**Comment by ncdc** [comment]
@fabriziopandini this doesn't look like a breaking change to me?

**Comment by fabriziopandini** [comment]
@ncdc this is not breaking any API, but there are users relying on the current file location for their `PreKubeadmCommands`, so I want to be sure they are properly informed.

**Comment by arghya88** [comment]
@vincepri should we then change the title of the PR to warning?

cc @fabriziopandini

**Comment by fabriziopandini** [comment]
+1 to add warning and retitle in something like "capk generates kubadm config files under /run/kubeadm/ instead of /tmp"

**Comment by vincepri** [comment]
Thanks for marking it as a breaking change!

**Comment by vincepri** [comment]
/ test pull-cluster-api-e2e-full-main

**Review by vincepri — APPROVED** [review]
/lgtm

**Pull request merged**